src/libs/sys-util-rs/src/fs.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2019-2021 Alibaba Cloud
+// Copyright (c) 2019-2021 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::ffi::OsString;
+use std::fs;
+use std::io::Result;
+use std::path::{Path, PathBuf};
+
+use crate::eother;
+
+const FUSE_SUPER_MAGIC: u32 = 0x65735546;
+
+/// Get bundle path (current working directory).
+pub fn get_bundle_path() -> Result<PathBuf> {
+    std::env::current_dir()
+}
+
+/// Get the basename of the canonicalized path
+pub fn get_base_name<P: AsRef<Path>>(src: P) -> Result<OsString> {
+    let s = src.as_ref().canonicalize()?;
+    s.file_name().map(|v| v.to_os_string()).ok_or_else(|| {
+        eother!(
+            "failed to get base name of path {}",
+            src.as_ref().to_string_lossy()
+        )
+    })
+}
+
+/// Check whether `path` is on a fuse filesystem.
+pub fn is_fuse_fs<P: AsRef<Path>>(path: P) -> bool {
+    if let Ok(st) = nix::sys::statfs::statfs(path.as_ref()) {
+        if st.filesystem_type().0 == FUSE_SUPER_MAGIC as i64 {
+            return true;
+        }
+    }
+    false
+}
+
+/// Check whether `path` is on a fuse filesystem.
+pub fn is_overlay_fs<P: AsRef<Path>>(path: P) -> bool {
+    if let Ok(st) = nix::sys::statfs::statfs(path.as_ref()) {
+        if st.filesystem_type() == nix::sys::statfs::OVERLAYFS_SUPER_MAGIC {
+            return true;
+        }
+    }
+    false
+}
+
+/// Check whether the given path is a symlink.
+pub fn is_symlink<P: AsRef<Path>>(path: P) -> std::io::Result<bool> {
+    let path = path.as_ref();
+    let meta = fs::metadata(path)?;
+
+    Ok(meta.file_type().is_symlink())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
```

Comment:
While I know the functions are simple, it'd be good to have unit tests for is_symlink, is_overlay_fs, is_fuse_fs??

Reply:
`is_symlink` could be tested easily enough.  `is_overlay_fs` and `is_fuse_fs` would be difficult to unit test; the test would have to set up real overlay or fuse filesystems on the host, which is not trivial without knowing the system's configuration.

Reply:
@egernst you are so smart, it actually disclose an implementation flaw:(
But it's hard to test fuse, so only add test cases for symlink and overlayfs.